tests/FINDOLOGIC/Export/Tests/XmlSerializationTest.php

```diff
@@ -605,12 +605,39 @@ public function testAllPricesCanBeSet(): void
         );
     }
 
-    public function testPricesAreNotInstancesOfPriceThrowsAnException()
+    public function testPricesAreNotInstancesOfPriceThrowsAnException(): void
     {
         $this->expectException(InvalidArgumentException::class);
         $this->expectExceptionMessage(sprintf('Given prices must be instances of %s', Price::class));
 
         $item = $this->exporter->createItem('123');
         $item->setAllPrices([new stdClass()]);
     }
+
+    public function testAddingMultipleAttributesWillNotOverrideOldOnes(): void
+    {
+        /** @var XMLItem $item */
+        $item = $this->getMinimalItem();
+
+        $expectedAttributes = ['orange', 'yellow', 'pink'];
+        $attr1 = new Attribute('color', ['orange', 'yellow', 'yellow']);
+        $attr2 = new Attribute('color', ['pink', 'orange', 'pink']);
+
+        $item->addAttribute($attr1);
+        $item->addAttribute($attr2);
+
+        $page = new Page(0, 1, 1);
+        $page->addItem($item);
+        $document = $page->getXml();
+        $xpath = new DOMXPath($document);
+
+        $values = $xpath->query('//attribute')->item(0)->childNodes->item(1)->childNodes;
+
+        $actualAttributes = [];
+        for ($i = 0; $i < $values->length; $i++) {
```

Comment:
Is it not possible to use `foreach` here?

Reply:
Apparently it is possible, I just didn't knew because it wasn't obvious while debugging:

![Selection_026](https://user-images.githubusercontent.com/20555112/66650745-a1959c00-ec31-11e9-8461-9877852f09e7.png)

Reply:
`DOMNodeList` implements `Traversable`, so that's why you can use `foreach`.

Reply:
I am just saying it wasn't obvious ;).